### PR TITLE
Fail request when error occurs during download

### PR DESCRIPTION
**Pull request by cesnietor**
fixes: https://github.com/minio/console/issues/3006

Checks whether the download was complete in the frontend.
In the backend, setting the headers doesn't work if it fails during io.Copy cause it already starterd writing to the body. So we just log it and handle it in the client layer.

On error:
After:
Screenshot 2024-01-24 at 5 06 38 PM

### Test Steps:
- Deploy a distributed minio e.g. using kind 4 nodes
- expose the service 
- Upload a big object e.g. 100Mb
- Go to the Browser UI
- Cordon one of the nodes
- Download the object (enabling throttling in your browser to emulate slow speed)
- Delete one of the minio pods (it won't start since we cordoned a node)
- In the UI the object download should eventually fail (before it was downloading an inclomplete object)

/// Other tests
- Test downloading multiple files (selecting multiple)
- Test downloading multiple files (selecting one, then download, then only other one and click download) so that the download box appears.

### CODE REVIEW

File: api/user_objects.go [hunk 1]
```diff
@@ -448,7 +448,8 @@ func getDownloadObjectResponse(session *models.Principal, params objectApi.Downl
 		// override filename is set
 		decodeOverride, err := base64.StdEncoding.DecodeString(*params.OverrideFileName)
 		if err != nil {
-			return
+			ErrorWithContext(ctx, fmt.Errorf("unable to decode OverrideFileName: %v", err))
+			panic(err)
```
Comment (prakashsvmx) [inline api/user_objects.go:452]: could we just log error instead of  `panic` ?

Comment (cesnietor) [reply api/user_objects.go:452]: panic is needed to cause a fail response @prakashsvmx , logging and just returning is what was causing the issue.

Comment (cesnietor) [reply api/user_objects.go]: After longer investigation I found that Copy already starts writing to the body so we can't send anything to the status or headers anymore. So we need to handle it at the client level.
But it an error happens before that, we can set the proper failure headers and statusCode.

**Review by prakashsvmx — APPROVED** [review]

**Review by bexsoft — APPROVED** [review]
Tested, LGTM
Screenshot 2024-01-26 at 3 05 44 p m

File: web-app/src/screens/Console/Buckets/ListBuckets/Objects/utils.ts [hunk 2]
```diff
@@ -143,7 +146,7 @@ export const download = (
             return;
           }
         }
-        errorCallback(`Unexpected response status code (${req.status}).`);
+        errorCallback(`Unexpected response, download incomplete.`);
```
Comment (vadmeste) [inline web-app/src/screens/Console/Buckets/ListBuckets/Objects/utils.ts:149]: Removing the status code does not help debugging, can we try to add more information here ?

Comment (cesnietor) [reply web-app/src/screens/Console/Buckets/ListBuckets/Objects/utils.ts:149]: status by itself was also not very helpful. You'll need to see the logs for now. We were not exposing the errors before either.
Exposing detailed errors is in our backlog.

**Pull request merged**